rbac/server/api/roles.py

```diff
@@ -171,6 +180,86 @@
     return json({"id": role_id, "description": role_description})
 
 
+@ROLES_BP.delete("api/roles/<role_id>")
+@authorized()
+async def delete_role(request, role_id):
+    """Delete a role by it's next_id.
+    Args:
+        role_id:
+            str: the role_id field of the targeted role
+    Returns:
+        json:
+            dict: {
+                message:
+                    str: the status of the role delete operation
+                deleted:
+                    int: count of the number of roles that were deleted
+            }
+    Raises:
+        ApiForbidden:
+            The user is not a system admin or owner of the targeted
+            role.
+        ApiNotFound:
+            The role does not exist in RethinkDB.
+        ApiInternalError:
+            There was an error compiling blockchain transactions.
+    """
+    txn_key, txn_user_id = await utils.get_transactor_key(request)
+
+    # does the role exist?
+    if not await roles_query.does_role_exist(request.app.config.DB_CONN, role_id):
+        LOGGER.warning(
+            "Nonexistent Role – User %s is attempting to delete the nonexistent role %s",
+            txn_user_id,
+            role_id,
+        )
+        return await handle_not_found(
+            request, ApiNotFound("The targeted role does not exist.")
+        )
+
+    is_role_owner = await check_role_owner_status(txn_user_id, role_id)
+    if not is_role_owner:
+        is_admin = await check_admin_status(txn_user_id)
+        if not is_admin:
+            LOGGER.warning(
+                "Permission Denied – User %s does not have sufficient privilege to delete role %s.",
+                txn_user_id,
+                role_id,
+            )
+            return await handle_errors(
+                request, ApiForbidden("You do not have permission to delete this role.")
+            )
+
+    txn_list = []
+    txn_list = await create_rjct_ppsls_role_txns(txn_key, role_id, txn_user_id, txn_list)
```

Comment:
A role's open proposals are still not closing after role has been deleted. I believe it is because of the function fetch_open_proposals_by_targe() is not getting all open proposals properly. Refer to comments for that function.